angular-workspace/projects/ni/nimble-angular/src/directives/table-column/anchor/nimble-table-column-anchor-router-link-with-href.directive.ts

```diff
@@ -0,0 +1,33 @@
+import { Directive, HostListener } from '@angular/core';
+import { RouterLinkWithHref } from '@angular/router';
+import type { TableColumnAnchorCellView } from '@ni/nimble-components/dist/esm/table-column/anchor/cell-view';
+
+/**
+ * Directive to handle nimble-table-column-anchor RouterLink support.
+ * Note: Clients need to use [nimbleRouterLink] instead of [routerLink], so that there
+ * won't also be an active RouterLink directive incorrectly handling navigation.
+ */
+@Directive({
+    selector: 'nimble-table-column-anchor[nimbleRouterLink]'
+})
+export class NimbleTableColumnAnchorRouterLinkWithHrefDirective extends RouterLinkWithHref {
+    @HostListener('delegated-event', ['$event.detail.originalEvent'])
+    private onDelegatedEvent(delegatedEvent: Event): void {
+        if (delegatedEvent.type !== 'click') {
+            return;
+        }
+
+        const clickEvent = delegatedEvent as MouseEvent;
+        const anchor = (delegatedEvent.target as TableColumnAnchorCellView).anchor;
+
+        if (!anchor) {
+            return;
+        }
+
+        // Let the router handle this navigation
+        this.routerLink = anchor.href;
```

Comment:
So we need the resulting string that comes off the anchor to be the raw href contents, ie. relative paths should look like `../` in order for the angular router to do it's [relative link magic](https://angular.io/api/router/RouterLink#relative-link-paths)

A normal anchor will do processing on the href value and we have a bug filed to nimble saying them not doing processing on the href value is a bug: https://github.com/microsoft/fast/issues/6591

Instead of digging into the TableColumnAnchorCellView's actual rendered UI to get the value applied to the view element, can you instead just ask the TableColumnAnchorCellView for it's original `href` property.

That way we are not dependent on buggy behavior / implementation details of the specific view's internal state.

Reply:
Done